libraries/abstractions/transport/secure_sockets/transport_secure_sockets.h

```diff
@@ -0,0 +1,188 @@
+
+    /**
+     * @brief Set this to a non-NULL value to use ALPN.
+     *
+     * This string must be NULL-terminated.
+     *
+     * See [this link]
+     * (https://aws.amazon.com/blogs/iot/mqtt-with-tls-client-authentication-on-port-443-why-it-is-useful-and-how-it-works/)
+     * for more information.
+     */
+    const char * pAlpnProtos;
+
+    /**
+     * @brief Length of the ALPN protocols array.
+     */
+    uint32_t alpnProtosLen;
+
+    /**
+     * @brief Disable server name indication (SNI) for a TLS session.
+     */
+    bool disableSni;
+
+    /**
+     * @brief Set this to a non-zero value to use TLS max fragment length
+     * negotiation (TLS MFLN).
+     *
+     * @note The network stack may have a minimum value for this parameter and
+     * may return an error if this parameter is too small.
+     */
+    size_t maxFragmentLength;
+
+    const char * pRootCa;     /**< @brief String representing a trusted server Root CA certificate. */
+    size_t rootCaSize;        /**< @brief Size associated with #IotNetworkCredentials_t.pRootCa. */
+    const char * pClientCert; /**< @brief String representing the client certificate. */
+    size_t clientCertSize;    /**< @brief Size associated with #IotNetworkCredentials_t.pClientCert. */
+    const char * pPrivateKey; /**< @brief String representing the client certificate's private key. */
+    size_t privateKeySize;    /**< @brief Size associated with #IotNetworkCredentials_t.pPrivateKey. */
+} SocketsConfig_t;
+
+
+/**
+ * @brief Sets up a TLS session on top of a TCP connection using the Secure Sockets API.
+ *
+ * @param[out] pNetworkContext The output parameter to return the created network context.
+ * @param[in] pServerInfo Server connection info.
+ * @param[in] pSocketsConfig socket configs for the connection.
+ *
+ * @return #TRANSPORT_SOCKET_STATUS_SUCCESS on success;
+ *         #TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, #TRANSPORT_SOCKET_STATUS_INSUFFICIENT_MEMORY,
+ *         #TRANSPORT_SOCKET_STATUS_INVALID_CREDENTIALS, #TRANSPORT_SOCKET_STATUS_INTERNAL_ERROR,
+ *         #TRANSPORT_SOCKET_STATUS_DNS_FAILURE, #TRANSPORT_SOCKET_STATUS_CONNECT_FAILURE on failure.
+ */
+TransportSocketStatus_t SecureSocketsTrasnport_Connect( NetworkContext_t * pNetworkContext,
+                                                        const ServerInfo_t * pServerInfo,
+                                                        const SocketsConfig_t * pSocketsConfig );
+
+/**
+ * @brief Closes a TLS session on top of a TCP connection using the Secure Sockets API.
+ *
+ * @param[out] pNetworkContext The output parameter to end the TLS session and
+ *             clean the created network context.
+ *
+ * @return #TRANSPORT_SOCKET_STATUS_SUCCESS on success;
+ *         #TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, #TRANSPORT_SOCKET_STATUS_INTERNAL_ERROR on failure.
+ */
+int32_t SecureSocketsTrasnport_Disconnect( const NetworkContext_t * pNetworkContext );
```

Comment:
Typo for `transport` word in all function names
```suggestion
int32_t SecureSocketsTransport_Disconnect( const NetworkContext_t * pNetworkContext );
```

Reply:
fixed in 637fc71df6534bebe6497a06ed7e87c6fb7c8028

Reply:
Should the input parameter be of non-const type (`NetworkContext_t * `) as the function would modify the value of the context?